pkg/services/store/entity_events.go

```diff
@@ -0,0 +1,138 @@
+func CreateDatabaseEntityId(internalId interface{}, orgId int64, entityType EntityType) string {
+	var internalIdAsString string
+	switch id := internalId.(type) {
+	case string:
+		internalIdAsString = id
+	default:
+		internalIdAsString = fmt.Sprintf("%#v", internalId)
+	}
+
+	return fmt.Sprintf("database/%d/%s/%s", orgId, entityType, internalIdAsString)
+}
+
+type EntityEvent struct {
+	Id        int64
+	EventType EntityEventType
+	EntityId  string
+	Created   int64
+}
+
+type SaveEventCmd struct {
+	EntityId  string
+	EventType EntityEventType
+}
+
+// EntityEventsService is a temporary solution to support change notifications in an HA setup
+// With this service each system can query for any events that have happened since a fixed time
+//go:generate mockery --name EntityEventsService --structname MockEntityEventsService --inpackage --filename entity_events_mock.go
+type EntityEventsService interface {
+	registry.BackgroundService
+	SaveEvent(ctx context.Context, cmd SaveEventCmd) error
+	GetLastEvent(ctx context.Context) (*EntityEvent, error)
+	GetAllEventsAfter(ctx context.Context, id int64) ([]*EntityEvent, error)
+
+	deleteEventsOlderThan(ctx context.Context, duration time.Duration) error
+}
+
+func ProvideEntityEventsService(cfg *setting.Cfg, sqlStore *sqlstore.SQLStore) EntityEventsService {
+	return &entityEventService{
+		sql: sqlStore,
+		log: log.New("entity-events"),
+	}
+}
+
+type entityEventService struct {
+	sql *sqlstore.SQLStore
+	log log.Logger
+}
+
+func (e *entityEventService) SaveEvent(ctx context.Context, cmd SaveEventCmd) error {
+	return e.sql.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		_, err := sess.Insert(&EntityEvent{
+			EventType: cmd.EventType,
+			EntityId:  cmd.EntityId,
+			Created:   time.Now().Unix(),
+		})
+		return err
+	})
+}
+
+func (e *entityEventService) GetLastEvent(ctx context.Context) (*EntityEvent, error) {
+	var entityEvent *EntityEvent
+	err := e.sql.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		bean := &EntityEvent{}
+		found, err := sess.OrderBy("id desc").Get(bean)
+		if found {
+			entityEvent = bean
+		}
+		return err
+	})
+
+	return entityEvent, err
+}
+
+func (e *entityEventService) GetAllEventsAfter(ctx context.Context, id int64) ([]*EntityEvent, error) {
+	var evs = make([]*EntityEvent, 0)
+	err := e.sql.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		return sess.OrderBy("id asc").Where("id > ?", id).Find(&evs)
+	})
+
+	return evs, err
+}
+
+func (e *entityEventService) deleteEventsOlderThan(ctx context.Context, duration time.Duration) error {
+	return e.sql.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		maxCreated := time.Now().Add(-duration)
+		deletedCount, err := sess.Where("created < ?", maxCreated.Unix()).Delete(&EntityEvent{})
+		e.log.Info("deleting old events", "count", deletedCount, "maxCreated", maxCreated)
+		return err
+	})
+}
+
+func (e *entityEventService) Run(ctx context.Context) error {
```

Comment:
Think `entityEventService` should implement [IsDisabled](https://github.com/grafana/grafana/blob/main/contribute/architecture/backend/services.md#disabled-services) method so that `Run` is not run when feature toggle is disabled.

Reply:
ahh awesome, I didn't know about that function! This reminded me about returning a dummy service when the feature flag is disabled - done both